### PR TITLE
WIP: reading order removal initial work

### PR DESCRIPTION
**Pull request by benleetownsend**

### CODE REVIEW

File: finetune/base_models/bert/featurizer.py [hunk 1]
```diff
@@ -8,6 +8,80 @@
...
+    has_improved = tf.cast(
+        tf.cond(
+            tf.equal((tf.train.get_global_step() - 1) % eval_freq, 0),
+            true_fn=lambda: tf.py_function(func=py_has_improved, inp=[], Tout=tf.bool),
+            false_fn=lambda: False
+        ),
+        tf.float32
+    )
+    decay = tf.get_variable("decay", shape=[], dtype=tf.float32, trainable=False, initializer=tf.zeros_initializer())
+    decay_op = tf.assign(decay, get_decay_for_half(total_num_steps) * has_improved + decay * (1 - has_improved))
```
Comment (madisonmay) [inline finetune/base_models/bert/featurizer.py:40]: might be preferable to increment one step at a time?  rather than a sharp jump once model gets lower loss at previous %?
